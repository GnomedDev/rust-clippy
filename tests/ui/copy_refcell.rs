#![warn(clippy::copy_refcell)]

use std::cell::RefCell;

struct MyStruct {
    field: RefCell<u8>,
}

fn main() {
    let local = RefCell::new(0_u8);
    let large_local = RefCell::new([0_u8; 1024]);
}
