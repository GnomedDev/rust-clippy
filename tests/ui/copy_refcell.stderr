error: `RefCell` used with a type that implements `Copy`
  --> tests/ui/copy_refcell.rs:6:12
   |
LL |     field: RefCell<u8>,
   |            ^^^^^^^^^^^
   |
   = help: replace the usage of `RefCell` with `Cell`, which does not have to track borrowing at runtime
   = note: `-D clippy::copy-refcell` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::copy_refcell)]`

error: `RefCell` used with a type that implements `Copy`
  --> tests/ui/copy_refcell.rs:10:17
   |
LL |     let local = RefCell::new(0_u8);
   |                 ^^^^^^^^^^^^^^^^^^
   |
   = help: replace the usage of `RefCell` with `Cell`, which does not have to track borrowing at runtime

error: aborting due to 2 previous errors

